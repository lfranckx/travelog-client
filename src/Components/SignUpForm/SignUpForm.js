import React, { Component } from 'react';
import AuthApiService from '../../Services/auth-api-service'
import ArticleContext from '../../Contexts/ArticleContext';

export default class SignUpForm extends Component {
    static defaultProps = {
        onSubmitSuccess: () => {}
    }

    static contextType = ArticleContext;

    state = { error: null };

    handleSubmit = ev => {
        ev.preventDefault();
        const { email, username, password, first_name, last_name } = ev.target;
        this.setState({ error: null });
        
        AuthApiService.postUser({
            email: email.value,
            username: username.value,
            password: password.value,
            first_name: first_name.value,
            last_name: last_name.value
        })
        .then(res => {
            AuthApiService.postLogin({
                username: username.value,
                password: password.value
            })
            .then(user => {
                username.value = ''
                password.value = ''
                this.props.onSignUpSuccess()
            })
        })
        .catch(res => {
            this.setState({ error: res.error })
        });
    }

    render() {
        const { error } = this.state;
        return (
            <form
                className="auth-form"
                onSubmit={this.handleSubmitJwtAuth}
            >
                <div role='alert'>{error && <p className='error'>{error}</p>}</div>
                <div className="input-box">
                    <label>Email</label>
                    <input
                        type="text" 
                        name='email' 
                        aria-label='email'
                        className='email' 
                        required
                    />
                </div>
                <div className="input-box">
                    <label>Username</label>
                    <input
                        type="text" 
                        name='username' 
                        aria-label='username'
                        className='username' 
                        required
                    />
                </div>
                <div className="input-box">
                    <label>Password</label>
                    <input
                        type="text" 
                        name='password' 
                        aria-label='password'
                        className='password' 
                        required
                    />
                </div>
                <div className="input-box">
                    <label>First Name</label>
                    <input
                        type="text" 
                        name='first-name' 
                        aria-label='first-name'
                        className='first-name' 
                        required
                    />
                </div>
                <div className="input-box">
                    <label>Last Name</label>
                    <input
                        type="text" 
                        name='last-name' 
                        aria-label='last-name'
                        className='last-name' 
                        required
                    />
                </div>
            </form>
        );
    }
}