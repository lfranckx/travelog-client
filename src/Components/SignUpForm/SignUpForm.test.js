import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SignUpForm from './SignUpForm';

describe(`SignUpForm component`, () => {
    const props = {
        handleSignUpSuccess: () => {}
    };

    it(`renders form given props`, () => {
        const wrapper = shallow(<SignUpForm {...props}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});