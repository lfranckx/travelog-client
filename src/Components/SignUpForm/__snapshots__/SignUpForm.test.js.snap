// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm component renders form given props 1`] = `
<form
  className="sign-up-form"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="name-box"
  >
    <div
      className="name-input"
    >
      <label>
        First Name
      </label>
      <input
        aria-label="first_name"
        className="first_name"
        name="first_name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="name-input"
    >
      <label>
        Last Name
      </label>
      <input
        aria-label="last_name"
        className="last_name"
        name="last_name"
        required={true}
        type="text"
      />
    </div>
  </div>
  <div
    className="name-box"
  >
    <div
      className="name-input"
    >
      <label>
        Email
      </label>
      <input
        aria-label="email"
        className="email"
        name="email"
        required={true}
        type="text"
      />
    </div>
    <div
      className="name-input"
    >
      <label>
        Username
      </label>
      <input
        aria-label="username"
        className="username"
        name="username"
        required={true}
        type="text"
      />
    </div>
  </div>
  <div
    className="password-box"
  >
    <label>
      Password
    </label>
    <input
      aria-label="password"
      className="password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <button
    className="submit"
    type="submit"
  >
    Register
  </button>
</form>
`;
